### PR TITLE
lesson03 all assignments

### PR DESCRIPTION
**Pull request by fortucj**
Had to work on these with no internet connection, so it's one big submission instead of frequent progressive commits.  Thanks for your patience.

### CODE REVIEW

File: students/fortucj/lesson03/list_lab.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+This is the List Lab.
+
+Entries are case irrelevant.
+All strings presented in title case.
+Item removal is executed for all instances of a repeated entry.
+Input for a repeated entry is presented once.
+"""
+
+
+def Series1():
```
Comment (charlesrobison) [inline students/fortucj/lesson03/list_lab.py]: Function names should be snake_case in Python

File: students/fortucj/lesson03/list_lab.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    zero-based indexing, so you will need to correct.
+    Add another fruit to the beginning of the list using “+” and display the list.
+    Add another fruit to the beginning of the list using insert() and display the list.
+    Display all the fruits that begin with “P”, using a for loop.
+    """
+
+    fruits = ['Apples', 'Pears', 'Oranges', 'Peaches']  # 1st and second bullets
+    print(fruits)
+
+    new_fruit1 = input("Please provide another fruit: ").title()  # 3rd & 4th bullets
```
Comment (charlesrobison) [inline students/fortucj/lesson03/list_lab.py]: Good use of title()

File: students/fortucj/lesson03/list_lab.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    fruits.append(new_fruit1)
+    print(fruits)
+
+    length = len(fruits)  # 5th bullet
+    "Please provide a number between 1 and {}: ".format(length)
```
Comment (charlesrobison) [inline students/fortucj/lesson03/list_lab.py]: Love that you used len() here!

File: students/fortucj/lesson03/list_lab.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+For each “no”, delete that fruit from the list.
+For any answer that is not “yes” or “no”, prompt the user to answer with one of those two \
+values (a while loop is good here)
+Display the list.
+    """
+
+    fruits3 = ask()  # call function to take default list or execute Series1
+
+    item_flag = []  # empty matrix allowing prevention of querying repeated fruits
+    fruits3_copy = fruits3[:]
```
Comment (charlesrobison) [inline students/fortucj/lesson03/list_lab.py]: Good!

File: students/fortucj/lesson03/mailroom.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+As the assignment becomes more complex, parts may be broken into separate functions.
+"""
+
+import sys
+
+# This was originally a list of lists.  I changed it based on the recommendation in the notes.
+data = [('Bob', [17.56]), ('Billy', [500.00, 1000.00]), ('Joe Schmoe', [2.00, 0.03, 45.00]), ('This Guy', [1.00, 100000]), ('That Gal', [9876.54])]
+
+# Generate a list of names to make looping easier.
+Name_List = []
```
Comment (charlesrobison) [inline students/fortucj/lesson03/mailroom.py]: Variables should be snake_case in Python - basically everything except class names.  name_list

File: students/fortucj/lesson03/mailroom.py [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+        if A_donation == 'quit':
+            Quit_Program()
+        if A_donation != 'none':
+            A_data[A_name_num][1].append(float(A_donation))  # add individual donation to original data
+            A_don_sum += float(A_donation)  # keep sum of donations
+    return A_data, A_don_sum  # return the updated original data and the sum of donations
+
+
+if __name__ == "__main__":
+    main()
```
Comment (charlesrobison) [inline students/fortucj/lesson03/mailroom.py]: Looking good but not able to add donation to new donor.  Did it work for you?

File: students/fortucj/lesson03/slicing_lab.py [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+test_tup = (1, 2.0, 3j, 'four', 1e-10, 'a', 6, 7, 'h', 8)
+test_tup9 = (1, 2.0, 3j, 'four', 1e-10, 'a', 6, 7, 'h')
+test_str11 = '1234567890e'
+assert first_last_exch(test_str) == 'tes this is a tesY'
+assert every_other(test_lis) == [1, 3j, 1e-10, 6, 'h']
+assert first4_last4(test_tup) == (1e-10,)
+assert elements_reversed(test_str) == 'tset a si siht seY'
+assert new_thirds(test_lis) == [7, 'h', 8, 1, 2.0, 3j, 'four', 1e-10, 'a', 6]
+assert new_thirds(test_tup9) == (6, 7, 'h', 1, 2.0, 3j, 'four', 1e-10, 'a')
+assert new_thirds(test_str11) == '890e1234567'
```
Comment (charlesrobison) [inline students/fortucj/lesson03/slicing_lab.py]: Looks good!

File: students/fortucj/lesson03/strformat_lab.py [hunk 8]
```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python
+
+"""
+string format lab
+
+Tasks 1-6
+"""
+
+
+def Task_One():
```
Comment (charlesrobison) [inline students/fortucj/lesson03/strformat_lab.py]: Make sure function names are snake_case: task_one()

File: students/fortucj/lesson03/mailroom.py (hunk 6 above)
Comment (fortucj) [reply students/fortucj/lesson03/mailroom.py]: Yes, it worked.  I tested it just now by copying/pasting the exact text in the file on GitHub, and ran it on my machine.  Running the file leads to the menu.  Option 1 lets you provide a new donor name.  Then you're prompted for as many donations as you'd like.  The result is email text.  Option 2 shows the updated donor list.  I tried sending you a video of me doing these steps but I believe the file is too big.

**Pull request merged**